ci/bazelisk.sh

```diff
@@ -7,26 +7,31 @@
 # but adds various flags to produce CI-friendly output. It does so by prociding a
 # command-line specified .bazelrc (that is applied alongside //.bazelrc).
 
-if [[ -n "$PWD_OVERRIDE" ]]; then
-    cd "$PWD_OVERRIDE" || exit
+if [[ -n "${PWD_OVERRIDE}" ]]; then
+    cd "${PWD_OVERRIDE}" || exit
 fi
 
 echo "Running bazelisk in $(pwd)."
 
 # An additional bazelrc must be synthesized to specify precisely how to use the
 # GCP bazel cache.
 GCP_BAZELRC="$(mktemp /tmp/XXXXXX.bazelrc)"
-trap 'rm $GCP_BAZELRC' EXIT
+trap 'rm ${GCP_BAZELRC}' EXIT
 
-if [[ -n "$GCP_BAZEL_CACHE_KEY" && -f "$GCP_BAZEL_CACHE_KEY" ]]; then
+if [[ -n "${GCP_BAZEL_CACHE_KEY}" && -f "${GCP_BAZEL_CACHE_KEY}" ]]; then
     echo "Applying GCP cache key; will upload to the cache."
-    echo "build --google_credentials=$GCP_BAZEL_CACHE_KEY" > "$GCP_BAZELRC"
+    echo "build --google_credentials=${GCP_BAZEL_CACHE_KEY}" >> "${GCP_BAZELRC}"
 else
     echo "No key/invalid path to key. Download from cache only."
-    echo "build --remote_upload_local_results=false" > "$GCP_BAZELRC"
+    echo "build --remote_upload_local_results=false" >> "${GCP_BAZELRC}"
 fi
 
+# Inject the OS version into a parameter used in the action key computation to
+# avoid collisions between different operating systems in the caches.
+# See #14695 for more information.
+echo "build --remote_default_exec_properties=OSVersion=\"$(lsb_release -ds)\"" >> "${GCP_BAZELRC}"
```

Comment:
Is there a plan to make the cache available to non-ci consumers like it was before by putting this into .bazelrc somewhere? If not could we document that the chache doesn't support non-ci users? I'd been using it experimentally until July, without enough instrumentation to see performance gains, but if I'm not mistaken, this is going to prevent cache hits.

Reply:
Yes this will prevent cache hits from anyone not running Bazel with `ci/bazelisk.sh`. On our website, we don't actually specify that the remote cache should be used. Rather, a [disk cache](https://docs.opentitan.org/doc/ug/bazel_notes/#disk-cache) is recommended.